deployment/copilot/devDeploy.sh

```diff
@@ -89,23 +89,22 @@ export CDK_DEPLOYMENT_STAGE=$STAGE
 export DOCKER_BUILDKIT=0
 export COMPOSE_DOCKER_CLI_BUILD=0
 
-if [ "$DESTROY_REGION" = true ] ; then
+if [ "$DESTROY_ENV" = true ] ; then
   set +e
   # Reset AWS_DEFAULT_REGION as the SDK used by Copilot will first check here for region to use to locate the Copilot app (https://github.com/aws/copilot-cli/issues/5138)
   export AWS_DEFAULT_REGION=""
```

Comment:
Minor: should we be preserving the user's current setting here? Something like:

```
CURRENT_DEFAULT_REGION=$AWS_DEFAULT_REGION
export AWS_DEFAULT_REGION=""

....

export AWS_DEFUALT_REGION=$CURRENT_DEFAULT_REGION
```

I also just realized that this isn't new to your PR, so not necessary to address now.